src/main/java/io/cryostat/net/web/http/api/beta/ProbeTemplateGetHandler.java

```diff
@@ -0,0 +1,157 @@
+
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import io.cryostat.core.agent.LocalProbeTemplateService;
+import io.cryostat.core.agent.ProbeTemplate;
+import io.cryostat.core.log.Logger;
+import io.cryostat.core.sys.FileSystem;
+import io.cryostat.messaging.notifications.NotificationFactory;
+import io.cryostat.net.AuthManager;
+import io.cryostat.net.security.ResourceAction;
+import io.cryostat.net.web.http.HttpMimeType;
+import io.cryostat.net.web.http.api.ApiVersion;
+import io.cryostat.net.web.http.api.v2.AbstractV2RequestHandler;
+import io.cryostat.net.web.http.api.v2.ApiException;
+import io.cryostat.net.web.http.api.v2.IntermediateResponse;
+import io.cryostat.net.web.http.api.v2.RequestParameters;
+
+import com.google.gson.Gson;
+import io.vertx.core.http.HttpMethod;
+
+public class ProbeTemplateGetHandler extends AbstractV2RequestHandler<String> {
+
+    static final String PATH = "probes";
+
+    private final Logger logger;
+    private final NotificationFactory notificationFactory;
+    private final LocalProbeTemplateService probeTemplateService;
+    private final FileSystem fs;
+    private final Gson gson;
+    private static final String NOTIFICATION_CATEGORY = "GetProbeTemplates";
+
+    @Inject
+    ProbeTemplateGetHandler(
+            AuthManager auth,
+            NotificationFactory notificationFactory,
+            LocalProbeTemplateService probeTemplateService,
+            Logger logger,
+            FileSystem fs,
+            Gson gson) {
+        super(auth, gson);
+        this.notificationFactory = notificationFactory;
+        this.logger = logger;
+        this.probeTemplateService = probeTemplateService;
+        this.fs = fs;
+        this.gson = gson;
+    }
+
+    @Override
+    public ApiVersion apiVersion() {
+        return ApiVersion.V2;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.GET;
+    }
+
+    @Override
+    public String path() {
+        return basePath() + PATH;
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    public boolean isOrdered() {
+        return true;
+    }
+
+    @Override
+    public boolean requiresAuthentication() {
+        return true;
+    }
+
+    @Override
+    public IntermediateResponse<String> handle(RequestParameters params) throws Exception {
+        List<String> probeTemplates = new ArrayList<>();
+        try {
+            for (ProbeTemplate p : this.probeTemplateService.getTemplates()) {
+                probeTemplates.add(
+                        "{\"name\" : \""
+                                + p.getFileName()
+                                + "\" , "
+                                + "\"xml\" : \""
+                                + p.serialize().replaceAll("\n", "").replaceAll("\"", "")
+                                + "\"}");
```

Comment:
Is this different from ex. `gson.toJson(p)`?

Reply:
Yea perfect, a `TypeAdapter` is exactly what I would suggest for a situation like that.